sycl/test-e2e/Basic/accessor/accessor.cpp

```diff
@@ -784,6 +784,40 @@ int main() {
     }
   }
 
+  // placeholder accessor exception (4)  // SYCL2020 4.7.6.9
+  {
+    sycl::queue q;
+    // host device executes kernels via a different method and there
+    // is no good way to throw an exception at this time.
+    sycl::range<1> r(4);
+    sycl::buffer<int, 1> b(r);
+    try {
+      using AccT = sycl::accessor<int, 1, sycl::access::mode::read_write,
+                                  sycl::access::target::device,
+                                  sycl::access::placeholder::true_t>;
+      AccT acc(b);
+
+      q.submit([&](sycl::handler &cgh) {
+        AccT acc2(b, cgh);
+        // Pass placeholder accessor to command, but having required a different
+        // accessor in the command. In past versions, we used to compare the
+        // number of accessors with the number of requirements, and if they
+        // matched, we did not throw, allowing this scenario that shouldn't be
+        // allowed.
+        cgh.single_task<class ph4>([=] { volatile int x = acc[0]; });
```

Comment:
why volatile?

Reply:
Wouldn't we want it to be volatile here? Otherwise the compiler might see `x` as a dead variable and in turn optimize out the access to `acc`. It may not happen, but I would feel safer _with_ the volatile qualifier here.

Reply:
In this case, if `x` was optimized out, the exception wouldn't be thrown, and we are expecting it, so the test would fail.

Reply:
I am not sure that is true, and if it is it would arguably be a bug. That said, it being optimized out would only mean that the kernel wouldn't have an argument for it, so we wouldn't set it. I expect we would still do our requirement checks and throw for this case, so maybe the example without volatile covers additional cases anyway. Let's roll with it. 👍